proto/substrait/algebra.proto

```diff
@@ -63,6 +63,17 @@ message ReadRel {
   Expression.MaskExpression projection = 4;
   substrait.extensions.AdvancedExtension advanced_extension = 10;
 
+  message PartitionColumns {
+    repeated ColumnType column_type = 1;
+    enum ColumnType {
+      COLUMN_TYPE_UNSPECIFIED = 0;
+      COLUMN_TYPE_NORMAL_COLUMN = 1;
+      COLUMN_TYPE_PARTITION_COLUMN = 2;
+    }
+  }
+
+  PartitionColumns partition_columns = 12;
+
```

Comment:
Don't we need to know the value of the column if it is `COLUMN_TYPE_PARTITION_COLUMN`?  For example:

```
  message PartitionColumns {
    message RegularColumn {}
    message PartitionColumn {
      Literal value;
    }
    message ColumnType {
      oneof column_type {
        RegularColumn regular = 0;
        PartitionColumn partition = 1;
      }
    }
    repeated ColumnType partition_columns = 0;
  }

  PartitionColumns partition_columns = 12;
```

Reply:
Thanks for your suggestion, and it should be better if Substrait plan can carry the value information. We currently extract that from the split paths in LocalFiles. Improved this PR accordingly.